src/related-manager.ts

```diff
@@ -244,6 +244,7 @@ class RelatedManager extends KalturaPlayer.core.FakeEventTarget {
       this.nextEntryTimeoutId = window.setTimeout(() => {
         this.playByIndex(0);
       }, seconds * 1000);
+      this.player.dispatchEvent(new KalturaPlayer.core.FakeEvent(RelatedEvent.RELATED_ENTRY_AUTO_PLAYED));
```

Comment:
i think the event should be fired from inside the setTimeout callback, because the timeout can be cancelled and you probably don't want to fire the event in that case

Reply:
the user behaviour  is the focus regardless of the outcome, the metrics (according to sagi  are focuses on the user behaviour/ intentions not the player outcomes / state)